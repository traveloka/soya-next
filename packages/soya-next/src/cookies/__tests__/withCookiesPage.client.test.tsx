import React from "react";
import { Cookies } from "react-cookie";
import { render } from "@testing-library/react";
import setupTest from "./utils/withCookiesPageSetup";

describe("withCookiesPage - Browser", () => {
  it("should add cookie to page props", async () => {
    const { PageWithCookies } = setupTest();
    const props = await PageWithCookies.getInitialProps({} as any);
    const { getByTestId } = render(
      <PageWithCookies {...props} cookies={new Cookies()} />
    );

    const withCookiesPropsDiv = getByTestId("with-cookies-props");
    expect(withCookiesPropsDiv.textContent).toMatchSnapshot();
  });
});
