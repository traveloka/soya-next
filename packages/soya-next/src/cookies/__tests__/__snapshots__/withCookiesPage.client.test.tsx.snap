// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withCookiesPage - Browser should add cookie to page props 1`] = `
"{
  \\"init\\": true,
  \\"cookies\\": {
    \\"changeListeners\\": [],
    \\"cookies\\": {},
    \\"HAS_DOCUMENT_COOKIE\\": true
  }
}"
`;
